drivers/wifi/nrf700x/zephyr/src/qspi/src/ficr_prog.c

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* @file
+ * @brief NRF Wi-Fi radio FICR programming functions
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr.h>
+#include "rpu_hw_if.h"
+#include "ficr_prog.h"
+
+
+LOG_MODULE_DECLARE(otp_prog, CONFIG_WIFI_LOG_LEVEL);
+
+static void write_word(unsigned int addr, unsigned int data)
+{
+	rpu_write(addr, &data, 4);
+}
+
+static void read_word(unsigned int addr, unsigned int *data)
+{
+	rpu_read(addr, data, 4);
+}
+
+unsigned int check_protection(unsigned int *buff, unsigned int off1, unsigned int off2,
+		    unsigned int off3, unsigned int off4)
+{
+	if ((buff[off1] == 0x00000000) &&
+		(buff[off2] == 0x00000000)  &&
+		(buff[off3] == 0x00000000) &&
+		(buff[off4] == 0x00000000))
+		return 0x00000000;
+	else if ((buff[off1] == 0xFFFFFFFF) &&
+		(buff[off2] == 0xFFFFFFFF) &&
+		(buff[off3] == 0xFFFFFFFF) &&
+		(buff[off4] == 0xFFFFFFFF))
+		return 0xFFFFFFFF;
+	else if ((buff[off1] == 0x50FA50FA) &&
+		(buff[off2] == 0x50FA50FA) &&
+		(buff[off3] == 0x50FA50FA) &&
+		(buff[off4] == 0x50FA50FA))
+		return 0x50FA50FA;
+	else
+		return 0xdeadbeef;
```

Comment:
Nice joke :D
Unfortunately we don't have any sense of humor.
Pleas us a sad predefined value here

Reply:
For all of returned values here.